tests/Application.Tests.Integration/Borrows/Commands/ApproveBorrowRequestTests.cs

```diff
@@ -0,0 +1,166 @@
+﻿using Application.Borrows.Commands;
+using Application.Common.Exceptions;
+using Application.Identity;
+using Domain.Statuses;
+using FluentAssertions;
+using Infrastructure.Persistence;
+using Microsoft.Extensions.DependencyInjection;
+using NodaTime;
+using Xunit;
+
+namespace Application.Tests.Integration.Borrows.Commands;
+
+public class ApproveBorrowRequestTests : BaseClassFixture
+{
+    public ApproveBorrowRequestTests(CustomApiFactory apiFactory) : base(apiFactory)
+    {
+        
+    }
+
+    [Fact]
+    public async Task ShouldApproveRequest_WhenRequestIsValid()
+    {
+        // Arrange
+        var document = CreateNDocuments(1).First();
+
+        var user = CreateUser(IdentityData.Roles.Employee, "abcdef");
+
+        var request = CreateBorrowRequest(user, document, BorrowRequestStatus.Pending);
+
+        await AddAsync(request);
+        
+        var command = new ApproveBorrowRequest.Command()
+        {
+            BorrowId = request.Id,
+        };
+        
+        // Act
+        var result = await SendAsync(command);
+        
+        // Assert 
+        result.Status.Should().Be(BorrowRequestStatus.Approved.ToString());
+        
+        // Cleanup
+        Remove(request);
+        Remove(user);
+        Remove(document);
+    }
+
+    [Fact]
+    public async Task ShouldThrowKeyNotFoundException_WhenRequestDoesNotExist()
+    {
+        // Arrange
+        var command = new ApproveBorrowRequest.Command()
+        {
+            BorrowId = Guid.NewGuid(),
+        };
+        
+        // Act
+        var result = async () => await SendAsync(command);
+
+        // Assert
+        await result.Should().ThrowAsync<KeyNotFoundException>()
+            .WithMessage("Borrow request does not exist.");
+    }
+
+    [Fact]
+    public async Task ShouldThrowConflictException_WhenDocumentIsLost()
```

Comment:
I would suggest asserting for if the result request is in NotProcessable state

Reply:
how would you assert the result if it throws an exception before returning anything?